cmd/ci-operator-prowgen/main.go

```diff
@@ -377,6 +377,9 @@ func generateJobs(
 		if configSpec.PromotionConfiguration != nil {
 			postsubmits[orgrepo] = append(postsubmits[orgrepo], *generatePostsubmitForTest("images", info, true, labels, generatePodSpec(info, "[images]", additionalPostsubmitArgs...)))
 		}
+		if configSpec.PrepublishConfiguration != nil {
+			presubmits[orgrepo] = append(presubmits[orgrepo], *generatePresubmitForTest("prepublish", info, generatePodSpec(info, "[prepublish]", additionalPresubmitArgs...)))
```

Comment:
Just a thought. For some reason I was expecting that we'll do the prepublishing within the `images` job, by appending the necessary `[prepublish]` target params.  But a separate job is also possible, I guess.

As a CI system, we don't care, both jobs will race for building the individual images and then the prepublish one will publish them.

I'm wondering about the effect on PR authors, though - having too many jobs may be confusing, especially if multiple ones fail on a single reason. I think separate job is worth having if its unique steps can fail in useful way - I would not expect prepublishing to fail unless we have infrastructure problems, which need admin intervention anyway, so maybe I would prefer to prepublish within the images job.

Reply:
Ah I didn't know each job could do more than one target. That makes sense. I think you've got a good point that this could easily be part of the images job.